flow/IndexedSet.h

```diff
@@ -1005,101 +1168,106 @@ void IndexedSet<T,Metric>::erase(iterator toErase) {
 
 // Returns x such that key==*x, or end()
 template <class T, class Metric>
+template <bool isConst>
 template <class Key>
-typename IndexedSet<T,Metric>::iterator IndexedSet<T,Metric>::find(const Key &key) const {
-	Node* t = root;
+typename IndexedSet<T, Metric>::template Impl<isConst>::IteratorT IndexedSet<T, Metric>::Impl<isConst>::find(
```

Comment:
nit: Instead of these explicit template instantiations, why not move the implementation inside the class definition, at least for the shorter methods, like this one?